src/software/ai/evaluation/calc_best_shot.cpp

```diff
@@ -43,15 +49,17 @@ namespace Evaluation
         // Calculate the best_shot based on what goal we're shooting at
         if (shoot_on_enemy_goal)
         {
-            best_shot =
-                calcBestShotOnGoal(field.enemyGoalpostNeg(), field.enemyGoalpostPos(),
-                                   point, obstacles, radius);
+            const Segment enemy_goal_segment =
+                Segment(field.enemyGoalpostNeg(), field.enemyGoalpostPos());
+            best_shot = std::make_optional(
```

Comment:
nit: Why not call `calcBestShotOnGoal` here? I originally built the functions on top of one another so it would hopefully be more understandable (in a mini-hierarchy).

Reply:
Changed to use calcBestShotOnGoal()

I added another overload that takes a vector<Circle> parameter instead of vector<Robot> and vector<Point> + radius